### PR TITLE
Implement make-config and edit-config commands in Lambo Zero

### PR DESCRIPTION
**Pull request by jonsugar**
Implement make-config and edit-config.

### Pre-amble and Assumptions
I'm following as closely as possible the patterns established with the existing code. It's possible there are instances where a simpler implementation could be used but I decided to favour existing convention. I figured that as I can't predict how you will think about my code, the safer option is to use existing precedence.

Thinking in a fail-fast-ish way, I decided to make a draft pull request even though it is functionally incomplete (still stable though) so that you can check progress and give feedback—I don't want to go too far down the rabbit hole for no reason.  Specifically, the creating and editing of the config file is finished but the reading and merging of the saved config with command line options is not finished yet.

I have tried to stay close to the shell based implementation of lambo but there there is one small difference. `make-config` and `edit-config` are just aliases of the same command that takes care of all the logic. I thought about just providing the one command but decided to keep the command-line experience the same.

I thought about more complicated features, for example an interactive wizard-like option when creating the initial config. In the end I decided to replicate existing functionality because: 1. That's what you did on stream. 2. It's probably quicker this way to get the php implementation released. Once stable and released, new fangled things can be considered.

### The $EDITOR Environment Variable
During my implementation I looked into using the $EDITOR command-line variable from within php but I came across several problems. 

In order to access the variable from within php I needed to edit my php.ini as the environment variables are not available by default. Even then, $EDITOR was not configured on my system. 

When I set `EDITOR=vim` in my shell environment and tried to open it via exec in php I received an error saying vim was not operating within a shell environment. It was basically unusable.

I decided attempting to use the shell editor was a non-starter. Even if it is set to sublime or vs code and would open properly the user would still need to mess about with php.ini.

TL;DR -- using the Mac 'open' utility, even with no application specified opens the config file in TextEdit.

### CODE REVIEW

File: app/Commands/NewCommand.php [hunk 1]
```diff
@@ -109,6 +114,11 @@ public function setConfig()
             return $this;
         });
```
Comment (jonsugar) [inline app/Commands/NewCommand.php]: I feel like this part here where I'm reading and merging the saved configuration could be extracted into its own class.

File: app/Actions/CreateOrEditConfig.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+{
+    public function __invoke()
+    {
+        $configPath = config('home_dir') . '/.lambo';
+        $configFilePath = "{$configPath}/config.json";
+
+        if (File::exists($configFilePath)) {
+            app('console')->info("Opening existing config file: {$configFilePath}");
+            $this->editConfig($configFilePath);
+            return 0;
```
Comment (mattstauffer) [inline app/Actions/CreateOrEditConfig.php]: Why return `0`?

File: app/Actions/CreateOrEditConfig.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        if (!File::exists($configPath)) {
+            File::makeDirectory($configPath);
+        }
+
+        File::put("$configFilePath", $this->configFileTemplate());
+        app('console')->info("Opening new config file {$configFilePath}");
+        $this->editConfig($configFilePath);
+    }
+
+    public function isMac()
```
Comment (mattstauffer) [inline app/Actions/CreateOrEditConfig.php]: Extract to a helper?

File: app/Actions/CreateOrEditConfig.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                $configFilePath
+            ));
+            return;
+        }
+        exec("open {$configFilePath}");
+    }
+
+    private function configFileTemplate(): string
+    {
+        return <<<'CONTENTS'
```
Comment (mattstauffer) [inline app/Actions/CreateOrEditConfig.php]: If we're doing json, let's instead make this an array and json encode it as we save

File: app/Commands/NewCommand.php [hunk 5]
```diff
@@ -12,18 +12,24 @@
...
 use LaravelZero\Framework\Commands\Command;
 
 class NewCommand extends Command
 {
     protected $signature;
     protected $description = 'Creates a fresh Laravel application';
+    /**
+     * @var array
+     */
+    private $savedConfig;
```
Comment (mattstauffer) [inline app/Commands/NewCommand.php]: can delete

File: app/Actions/CreateOrEditConfig.php (hunk 2 above)
Comment (jonsugar) [reply app/Actions/CreateOrEditConfig.php]: Just being a good command line citizen. I guess it's not totally required.

File: app/Actions/CreateOrEditConfig.php (hunk 4 above)
Comment (jonsugar) [reply app/Actions/CreateOrEditConfig.php]: redundant reply since you did that on stream 😆

File: app/Actions/CreateOrEditConfig.php (hunk 3 above)
Comment (jonsugar) [reply app/Actions/CreateOrEditConfig.php]: Yep. Will wait until you finish committing.

**Comment by jonsugar** [comment]
Closing so I can re-create the PR against the pr/83 branch
